src/Components/Web.JS/src/Rendering/EventForDotNet.ts

```diff
@@ -204,8 +211,38 @@ function parseMouseEvent(event: MouseEvent) {
   };
 }
 
-function isCheckbox(element: Element | null) {
-  return element && element.tagName === 'INPUT' && element.getAttribute('type') === 'checkbox';
+function isCheckbox(element: Element | null): boolean {
+  return !!element && element.tagName === 'INPUT' && element.getAttribute('type') === 'checkbox';
+}
+
+const timeBasedInputs = [
+  'date',
+  'datetime-local',
+  'month',
+  'time',
+  'week',
+];
+
+function isTimeBasedInput(element: Element | null): boolean {
+  return !!element && timeBasedInputs.indexOf(element.getAttribute('type') || '') !== -1;
+}
+
+function normalizeTimeBasedValue(element: Element): string {
+  const value = element['value'] as string;
+  const type = element['type'];
+  switch (type) {
+    case 'date':
+    case 'datetime-local':
+    case 'month':
+      return value;
+    case 'time':
+      return value.split(':').length === 2 ? value + ':00' : value;
```

Comment:
Also I know this is nitpicky, but we could avoid the `split` and its allocations:

```suggestion
      return value.length === 5 ? value + ':00' : value; // Convert hh:mm to hh:mm:00
```